File: Data Structure/Lab 4/DelPosi.c
// Write a program to delete a number from an array that is already sorted in an 
//ascending order. 

#include<stdio.h>
void main(){
    int n,a[n + 1],i = 0,index,val;

    printf("Enter size of an aay(n) :");
    scanf("%d",&n);

    printf("Enter value :");
    scanf("%d",&val);

    for(i = 0; i < n ; i++){
        printf("Enter element :");
        scanf("%d",&a[i]);
    }

    index = n;
    for(i = 0 ; i < n; i++){
        if(a[i] == val){
            index = i;
            break;
        }
    }

    if(index < n){
        for(i = index ; i < n - 1; i++){
            a[i] = a[i + 1];
        }
        n--;

        for(i = 0; i < n ; i++){
            printf("%d\n ,",a[i]);
        }
    }

    else{
        printf("Element not Found");
    }
}

File: Data Structure(Java)/Lab 3/DelPosi.java
//Write a program to delete a number from a given location in an array. 

import java.util.Scanner;

public class DelPosi{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter size of an array :");
        int n = sc.nextInt();
        int[] arr = new int[n];

        for(int i = 0 ; i < n ; i++){
            System.out.println("Enter array element     `1at "+i);
            arr[i] = sc.nextInt();
        }

        System.out.println("Final array :");
        for(int i = 0; i < n ; i++){
            if(i == n - 1){
                System.out.print(arr[i]);
            }
            else{
                System.out.print(arr[i]+",");
            }
        }

        System.out.println("Enter position where you want to delete element(0 - "+(n - 1)+") :");
        int posi = sc.nextInt();

        for(int i = posi; i < n - 1 ; i++){
               arr[i] = arr[i + 1];
        }

        n--;//So that it will print all elements

        for(int i = 0 ; i < n ; i++){
            if(i == n - 1){
                System.out.println(arr[i]);
            }
            else{
                System.out.print(arr[i]+",");
            }
        }

        sc.close();
    }
}